config.go

```diff
@@ -3,83 +3,91 @@ package main
 import (
 	"time"
 
-	"github.com/BurntSushi/toml"
+	"github.com/kelseyhightower/envconfig"
 )
 
 type ListenerUrlsConfig struct {
-	Urls []string `toml:"urls"`
+	Urls []string `envconfig:"URLS"`
 }
 
 type HAproxyConfig struct {
-	ReloadCmd    string `toml:"reload_command"`
-	VerifyCmd    string `toml:"verify_command"`
-	BindIP       string `toml:"bind_ip"`
-	TemplateFile string `toml:"template_file"`
-	ConfigFile   string `toml:"config_file"`
-	PidFile      string `toml:"pid_file"`
-	Disable      bool   `toml:"disable"`
-	User         string `toml:"user"`
-	Group        string `toml:"group"`
-	UseHostnames bool   `toml:"use_hostnames"`
+	ReloadCmd    string `envconfig:"RELOAD_COMMAND"`
+	VerifyCmd    string `envconfig:"VERIFY_COMMAND"`
+	BindIP       string `envconfig:"BIND_IP" default:"192.168.168.168"`
+	TemplateFile string `envconfig:"TEMPLATE_FILE" default:"views/haproxy.cfg"`
+	ConfigFile   string `envconfig:"CONFIG_FILE" default:"/etc/haproxy.cfg"`
+	PidFile      string `envconfig:"PID_FILE" default:"/var/run/haproxy.pid"`
+	Disable      bool   `envconfig:"DISABLE"`
+	User         string `envconfig:"USER" default:"haproxy"`
+	Group        string `envconfig:"GROUP" default:"haproxy"`
+	UseHostnames bool   `envconfig:"USE_HOSTNAMES" default:"false"`
 }
 
 type ServicesConfig struct {
-	NameMatch    string `toml:"name_match"`
-	ServiceNamer string `toml:"namer"`
-	NameLabel    string `toml:"name_label"`
+	NameMatch    string `envconfig:"NAME_MATCH"`
+	ServiceNamer string `envconfig:"NAMER" default:"docker_label"`
+	NameLabel    string `envconfig:"NAME_LABEL" default:"ServiceName"`
 }
 
 type SidecarConfig struct {
-	ExcludeIPs           []string `toml:"exclude_ips"`
-	Discovery            []string `toml:"discovery"`
-	StatsAddr            string   `toml:"stats_addr"`
-	PushPullInterval     duration `toml:"push_pull_interval"`
-	GossipMessages       int      `toml:"gossip_messages"`
-	LoggingFormat        string   `toml:"logging_format"`
-	LoggingLevel         string   `toml:"logging_level"`
-	DefaultCheckEndpoint string   `toml:"default_check_endpoint"`
+	ExcludeIPs           []string      `envconfig:"EXCLUDE_IPS" default:"192.168.168.168"`
+	Discovery            []string      `envconfig:"DISCOVERY" default:"docker"`
+	StatsAddr            string        `envconfig:"STATS_ADDR"`
+	PushPullInterval     time.Duration `envconfig:"PUSH_PULL_INTERVAL" default:"20s"`
+	GossipMessages       int           `envconfig:"GOSSIP_MESSAGES" default:"15"`
+	LoggingFormat        string        `envconfig:"LOGGING_FORMAT"`
+	LoggingLevel         string        `envconfig:"LOGGING_LEVEL" default:"info"`
+	DefaultCheckEndpoint string        `envconfig:"DEFAULT_CHECK_ENDPOINT" default:"/version"`
 }
 
 type DockerConfig struct {
-	DockerURL string `toml:"docker_url"`
+	DockerURL string `envconfig:"URL" default:"unix:///var/run/docker.sock"`
```

Comment:
`DOCKER_URL`?

Reply:
Same as above, section is called `docker` so it would be `DOCKER_DOCKER_URL`